### DISCUSSION TITLE
Undoing a migration

### DISCUSSION THREAD

**Discussion by masken98** [prisma/prisma#19054] [Q&A]
Hello,
I am using Prisma for a project now in school now. In doing so, I seem to have accidentally made a duplicate migration. Somehow this migration didn't fail so --rollback doesn't work but from what I can tell there isn't another way to go about reverting a change. Surely this is incorrect and I am just missing something. 
Cheers

**Comment by nurul3101** [comment]
Hello @masken98 👋 

Did you had a chance to look at generating [down migrations](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/generating-down-migrations)?

**Reply by masken98** [reply]
I did! I am a bit unclear of the usage of diff though. I have never defined a shadow database and such it has no URL but trying to run diff with --to-schema-datasource only gave errors like "error: Error validating: This line is invalid. It does not start with any known Prisma schema keyword."

**Reply by nurul3101** [reply]
Can you share the entire command that you are running?

**Reply by nurul3101** [reply]
Do let us know in case you still need any assistance.
